src/wsproto/handshake.py

```diff
@@ -236,14 +236,6 @@ def _process_connection_request(  # noqa: MC0001
             raise RemoteProtocolError(
                 "Missing header, 'Upgrade: WebSocket'", event_hint=RejectConnection()
             )
-        if version is None:
```

Comment:
This is dead code - there is another version check above.

Reply:
Let's run this through:

* init to `version = None`
* conditional `version = value`
* `if version != WEBSOCKET_VERSION:`
  - version could be `None` which is `!=` to `WEBSOCKET_VERSION` and triggers `RemoteProtocolError`
  - version could be `b'something'` which is `!=` to `WEBSOCKET_VERSION` and triggers `RemoteProtocolError`
  - version could be `b'13'` which is `==` to `WEBSOCKET_VERSION` and does not trigger an error
* `if version is None`
  - this wasn't already covered above - therefore cannot happen

Anything missing?

Reply:
Ah, yea it never worked. I think it should though, so how about moving the `if version is None` up before the `if version != WEBSOCKET_VERSION:` check? To be clear I think if version is None the error should hint at a 400 response, whereas if the version doesn't match I think it should hint at a 426.

Reply:
Good point - done.